src/distributions/integer.rs

```diff
@@ -13,46 +13,52 @@
 use {Rng};
 use distributions::{Distribution, Standard};
 
-impl Distribution<isize> for Standard {
-    #[inline]
-    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> isize {
-        rng.gen::<usize>() as isize
-    }
-}
-
-impl Distribution<i8> for Standard {
+impl Distribution<u8> for Standard {
     #[inline]
-    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> i8 {
-        rng.next_u32() as i8
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> u8 {
+        rng.next_u32() as u8
     }
 }
 
-impl Distribution<i16> for Standard {
+impl Distribution<u16> for Standard {
     #[inline]
-    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> i16 {
-        rng.next_u32() as i16
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> u16 {
+        rng.next_u32() as u16
     }
 }
 
-impl Distribution<i32> for Standard {
+impl Distribution<u32> for Standard {
     #[inline]
-    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> i32 {
-        rng.next_u32() as i32
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> u32 {
+        rng.next_u32()
     }
 }
 
-impl Distribution<i64> for Standard {
+impl Distribution<u64> for Standard {
     #[inline]
-    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> i64 {
-        rng.next_u64() as i64
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> u64 {
+        rng.next_u64()
     }
 }
 
 #[cfg(feature = "i128_support")]
-impl Distribution<i128> for Standard {
+impl Distribution<u128> for Standard {
     #[inline]
-    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> i128 {
-        rng.gen::<u128>() as i128
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> u128 {
+        if rng.bytes_per_round() < 128 {
+            // Use LE; we explicitly generate one value before the next.
+            let x = rng.next_u64() as u128;
+            let y = rng.next_u64() as u128;
+            (y << 64) | x
+        } else {
+            let mut val = 0u128;
+            unsafe {
+                let ptr = &mut val;
+                let b_ptr = &mut *(ptr as *mut u128 as *mut [u8; 16]);
+                rng.fill_bytes(b_ptr);
+            }
+            val.to_le()
```

Comment:
It would be nice to see benchmarks for this on a BE platform

Reply:
I sometimes just change things to `to_be()` and measure on x86_64. But it really only starts to show improvements for things like my SIMD experiment (twice as fast there).